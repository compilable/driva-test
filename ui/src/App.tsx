import React, { useEffect } from 'react';
import 'semantic-ui-css/semantic.min.css'
import UserDetailsFormOne from "./components/quoteForm/userDetailsFormOne"
import UserDetailsFormTwo from "./components/quoteForm/userDetailsFormTwo"
import { Route, Routes, useNavigate } from "react-router-dom"
import { getCustomErrorMessage, MainContainer, setFormData } from "./utils/util_v1.0"

export const AppWrapper = () => {

  const navigate = useNavigate()

  type processFormRequest = {
    formName: string,
    data: JSON
  }

  function processForm(data: processFormRequest) {
    console.log(data)

    if (data.formName === "q1") {
      // store data into localstore
      setFormData(data.formName, data.data)
      return navigate("q2")
    } else if (data.formName === "q2") {
      // store data into localstore
      return navigate("/")
    }

  }

  return (
    <Routes>
      <Route path="/" element={<UserDetailsFormOne formName={"q1"} showCustomError={getCustomErrorMessage} processForm={processForm} />} />
      <Route path="q2" element={<UserDetailsFormTwo formName={'q2'} showCustomError={getCustomErrorMessage} processForm={processForm} />} />
    </Routes>
  )
}


export const App = () => {

  useEffect(() => {
    // reset the 
    setFormData()
  }, []);


  return (
    <MainContainer>
      <AppWrapper />
    </MainContainer>
  )
};


