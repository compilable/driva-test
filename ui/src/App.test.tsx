import React from 'react';
import { render } from '@testing-library/react'
import { shallow, mount, ReactWrapper } from 'enzyme';
import toJSON from "enzyme-to-json"


import { App, AppWrapper } from "./App"
import { BrowserRouter } from 'react-router-dom';

test("has the main container loaded", () => {
    const app = shallow(<App />);
    expect(toJSON(app)).toMatchSnapshot()
});


test("has the wrapper loaded", () => {
    const wrapper = shallow(
        <BrowserRouter>
            <AppWrapper />
        </BrowserRouter>
    );
    expect(toJSON(wrapper)).toMatchSnapshot()
});