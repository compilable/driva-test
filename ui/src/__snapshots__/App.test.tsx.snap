// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has the main component loaded 1`] = `
<styled.div>
  <AppWrapper />
</styled.div>
`;

exports[`has the main container loaded 1`] = `
<styled.div>
  <AppWrapper />
</styled.div>
`;

exports[`has the wrapper loaded 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "back": [Function],
      "block": [Function],
      "createHref": [Function],
      "forward": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <AppWrapper />
</Router>
`;
